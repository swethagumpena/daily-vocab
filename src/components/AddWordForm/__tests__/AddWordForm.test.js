/* eslint-disable react/jsx-props-no-spreading */
import {
  render, screen, fireEvent, act,
} from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';
import React from 'react';
import AddWord from '../AddWordForm';

describe(AddWord.name, () => {
  afterEach(() => {
    jest.clearAllMocks();
  });
  const mockProps = {
    addNewVocab: jest.fn(),
  };

  test('should match snapshot', () => {
    const { container } = render(<AddWord {...mockProps} />);
    expect(container).toMatchSnapshot();
  });

  test('should display add a new word', () => {
    render(<AddWord {...mockProps} />);
    expect(screen.getByText('add a new word'));
  });

  test('should display a tick button', () => {
    render(<BrowserRouter><AddWord {...mockProps} /></BrowserRouter>);
    const addElement = screen.getByText('âœ“');
    expect(addElement.tagName).toBe('BUTTON');
    act(() => { fireEvent.click(addElement); });
    // expect(document.location.href).toBe('http://localhost/home');
  });
});
