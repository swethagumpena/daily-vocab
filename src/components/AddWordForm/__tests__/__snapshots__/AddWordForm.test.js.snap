// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddWord should match snapshot 1`] = `
<div>
  <div
    class="formContainer"
  >
    <p
      class="heading"
    >
      add a new word
    </p>
    <div
      class="formData"
    >
      <form>
        <div
          class="formItems"
        >
          <label
            class="wordSentence"
            for="word"
          >
            <p>
              What's the word:
            </p>
            <input
              name="word"
              placeholder="new word"
              type="text"
            />
          </label>
          <label
            class="wordSentence"
            for="sentence"
          >
            <p>
              Let's use it in a sentence:
            </p>
            <input
              name="sentence"
              placeholder="new sentence"
              type="text"
            />
          </label>
        </div>
        <button
          class="submitBtn"
          type="submit"
        >
          âœ“
        </button>
      </form>
    </div>
  </div>
</div>
`;
